### ISSUE TITLE
Add other style properties to Box and Flex

### ISSUE THREAD

**Issue by mannycarrera4** [Workday/canvas-kit#1423]
## 🚀 Feature Proposal

A clear and concise description of what the feature is.

## Motivation

When `Box` and `Flex` where initially created we added base style properties which have covered most use cases. As Canvas Kit grows, we're finding more and more use cases where other style properties will be nice to have such as `animation` textDecoration`. Chakra has a list of `Other Props` that we can probably reference

https://chakra-ui.com/docs/features/style-props#other-props

**Comment by vibdev** [comment]
Thoughts on adding psudo states too?  e.g.
```
<a color='blue' color_hover='green' color_focus='red'>Link</a>
```
I could see that being really handy, but could also see it being an explosion of props to add.

**Comment by mannycarrera4** [comment]
@vibdev yep! We've definitely talked about this as well! I believe @alanbsmith had some thoughts on this as well

**Comment by alanbsmith** [comment]
We've talked about this a lot internally. If we decided to add style props for pseudo classes, we'd likely follow [Chakra's pattern](https://chakra-ui.com/docs/features/style-props#pseudo). The debate has been whether to continue to invest in these components or spend that time figuring out what's next after Emotion.

It's something I'd love to add, and something I think these components really need (along with responsive style support). It's been hard to justify with the given timeline.
